portfolio/src/main/java/com/google/sps/utility/ValidateInput.java

```diff
@@ -15,53 +15,139 @@
 package com.google.sps.utility;
 
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 /** A class that contains methods to validate input */
 public final class ValidateInput {
 
   /** 
-   * Returns the number of comments shown entered by the user, or -1 if the 
-   * comment was invalid. Min must be greater than -1 and Max must be greater 
-   * than or equal to min 
+   * Returns the String input entered by the user, or throws an exception if
+   * the input was invalid.
+   * Min must be greater than -1 and Max must be greater 
+   * than or equal to min or else an exception is thrown.
    *
    * @param request the request received from the form that contains user input
    * @param parameter the name of the input parameter one is retreiving
    * @param min used to establish the lower bound of the input
    * @param max used to establish the upper bound of the input
    * @return the user's input (number) or -1 if it does not follow guidelines
    */
-  public int getUserNum(HttpServletRequest request, String parameter, int min, int max) {
+  static String getUserInput(HttpServletRequest request, String parameter, 
+      int min, int max) throws Exception {
     if (min <= -1) {
-      System.err.println("Min (" + min + ") must be greater than -1 ");
-      return -1;
+      String error = "Min (" + min + ") must be greater than -1 ";
+      System.err.println(error);
+      throw new Exception(error);
```

Comment:
Throwing a generic Exception is fine for now, but down the road we'll focus on throwing specific exceptions for specific error conditions, for example, creating a ValidationException class to indicate when a validation failure has occurred.

(Don't worry about making this change now - just something to keep in mind)